sntools/interaction_channels/o16nc_n.py

```diff
@@ -0,0 +1,116 @@
+    return f
+
+class Channel(BaseChannel):
+    def generate_event(self, eNu, dirx, diry, dirz):
+        """Return an event with the appropriate incoming/outgoing particles.
+
+        Input:
+            eNu: neutrino energy
+            dirx, diry, dirz: direction of outgoing particle (normalized to 1)
+        """
+        # Note: `self.flavor` is set during __init__
+        nu_flv = {'e': 12, 'eb': -12, 'x': 14, 'xb': -14}[self.flavor]
+
+        eE = self.get_eE(eNu, dirz)
+
+        evt = Event(2008016 if nu_flv > 0 else -2008016)
+        evt.incoming_particles.append([nu_flv, eNu, 0, 0, 1])  # incoming nu
+        evt.incoming_particles.append((8016, 14900, 0, 0, 1))  # oxygen-16 nucleus at rest
+        evt.outgoing_particles.append([2112, eE, dirx, diry, dirz])  # emitted neutron
+        # evt.outgoing_particles.append([nu_flv, eNu-e_thr, 0, 0, 1])  # outgoing nu
+        return evt
+    
+    def bounds_eE(self, eNu, *args):
+        """Return kinematic bounds for integration over eE.
+
+        Input:
+            eNu:  neutrino energy (in MeV)
+            args: [ignore this]
+        Output:
+            list with minimum & maximum allowed energy of outgoing (detected) particle
+        """
+        return [self.get_eE(eNu) - epsilon, self.get_eE(eNu) + epsilon]
+        
+    def get_eE(self, eNu, cosT=0):
+        """Return energy (in MeV) of outgoing (detected) particle.
+
+        Input:
+            eNu:  neutrino energy (in MeV)
+            cosT: cosine of the angle between neutrino and outgoing (detected) particle
+        """
+        eE = random.random()*(eNu-e_thr) + mN       
+        return eE
+    
+    def dSigma_dE(self, eNu, eE):
+        """Return differential cross section in MeV^-2.
+
+        Inputs:
+            eNu: neutrino energy
+            eE:  energy of outgoing (detected) particle
+        """
+        if eNu < e_thr or abs(self.get_eE(eNu) - eE) > epsilon:
+            # This should never happen, since we set bounds for eE and eNu accordingly above
+            # ... but just in case:
+            return 0
+
+        sigma = fit()(eNu)*10**(-42)  # cross-section at eNu from the interp1d fit 
+                                                # of Suzuki et al. 2018 data (units cm^2)
+        sigma *= (5.067731E10)**2  # convert cm^2 to MeV^-2: 
+                   # http://www.wolframalpha.com/input/?i=cm%2F(hbar+*+c)+in+MeV%5E(-1)
+        return sigma / (2 * epsilon)  # Ensure that integration over eE yields sigma
+
+    def dSigma_dCosT(self, eNu, cosT):
+        """Return differential cross section in MeV^-2 as a function of the 
+        emission angle of the outgoing (detected) particle.
+
+        Input:
+            eNu:  neutrino energy (MeV)
+            cosT: cosine of the angle between neutrino and outgoing (detected) particle
+        """
+        # Energy dependence is unclear, so we use a constant value for now.
+        if abs(cosT) > 1:
+            return 0
+        return 0.5
```

Comment:
This should be `0.5 * sigma` (where sigma is defined as in `dSigma_dE` above), to ensure that integrating `dSigma_dCosT` over `cosT` gives the physically correct value.

(Right now, this shouldn’t affect the output of sntools, since that’s only affected by the cosT-dependence and not the absolute value of this function. But I’d rather be physically accurate right away, so this doesn’t cause issues in the future.)